pkg/azure/azure.go

```diff
@@ -0,0 +1,231 @@
+)
+
+type ImageDescription struct {
+	GalleryName       string
+	GalleryImageName  string
+	Offer             string
+	Publisher         string
+	ResourceGroupName string
+	SKU               string
+}
+
+func NewImageDescription(
+	galleryName string,
+	galleryImageName string,
+	offer string,
+	publisher string,
+	resourceGroupName string,
+	sku string,
+) (*ImageDescription, error) {
+	return &ImageDescription{
+		GalleryName:       galleryName,
+		GalleryImageName:  galleryImageName,
+		Offer:             offer,
+		Publisher:         publisher,
+		ResourceGroupName: resourceGroupName,
+		SKU:               sku,
+	}, nil
+}
+
+type Credentials struct {
+	ID       string
+	Secret   string
+	TenantID string
+}
+
+func NewCredentials(clientID string, clientSecret string, tenantID string) (*Credentials, error) {
+	return &Credentials{
+		ID:       clientID,
+		Secret:   clientSecret,
+		TenantID: tenantID,
+	}, nil
+}
+
+func createGalleryImage(
+	ctx context.Context,
+	cred azcore.TokenCredential,
+	description *ImageDescription,
+	location string,
+	subscriptionID string,
+) (*armcompute.GalleryImage, error) {
+	galleryImageClient := armcompute.NewGalleryImagesClient(subscriptionID, cred, nil)
+
+	pollerResp, err := galleryImageClient.BeginCreateOrUpdate(
+		ctx,
+		description.ResourceGroupName,
+		description.GalleryName,
+		description.GalleryImageName,
+		armcompute.GalleryImage{
+			Location: to.StringPtr(location),
+			Properties: &armcompute.GalleryImageProperties{
+				OSType:           armcompute.OperatingSystemTypesLinux.ToPtr(),
+				OSState:          armcompute.OperatingSystemStateTypesGeneralized.ToPtr(),
+				HyperVGeneration: armcompute.HyperVGenerationV2.ToPtr(),
+				Identifier: &armcompute.GalleryImageIdentifier{
+					Offer:     to.StringPtr(description.Offer),
+					Publisher: to.StringPtr(description.Publisher),
+					SKU:       to.StringPtr(description.SKU),
+				},
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to create gallery image %s in %s: %w",
+			description.GalleryImageName,
+			description.GalleryName,
+			err,
+		)
+	}
+
+	resp, err := pollerResp.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("error creating gallery image %s in %s: %w",
+			description.GalleryImageName,
+			description.GalleryName,
+			err,
+		)
+	}
+
+	return &resp.GalleryImage, nil
+}
+
+func createGallery(
+	ctx context.Context,
+	cred azcore.TokenCredential,
+	description *ImageDescription,
+	location string,
+	subscriptionID string,
+) (*armcompute.Gallery, error) {
+	galleriesClient := armcompute.NewGalleriesClient(subscriptionID, cred, nil)
+
+	pollerResp, err := galleriesClient.BeginCreateOrUpdate(
+		ctx,
+		description.ResourceGroupName,
+		description.GalleryName,
+		armcompute.Gallery{
+			Location: to.StringPtr(location),
+			Properties: &armcompute.GalleryProperties{
+				Description: to.StringPtr("This is gallery description."),
```

Comment:
Is this the correct string?

Reply:
Lulz, it is _a_ correct string ;). Updated to `DKP Gallery`.